[POS Settings] Design updates

iamgabrielma opened the pull request:
Closes WOOMOB-1162

## Description
This PR updates the UI for POS Settings following the wireframes on pdfdoF-7RJ-p2 and existing POS styles for overall consistency. Note that we do not have final designs yet, what's in this PR will most likely be final for releasing i1. Initial designs have just been published at pdfdoF-80C-p2 so I took loose inspiration for the details while are being discussed and polished. 

Android's UI can be seen in the project update for further reference if needed.

Split view and large font sizes are not handled in this PR, will be done separately.

| Store light | Store dark |
|--------|--------|
| <img width="2266" height="1488" alt="Simulator Screenshot - iPad mini (A17 Pro) - US store - 2025-08-31 at 21 19 10" src="https://github.com/user-attachments/assets/cb40b56a-3158-4a55-9b71-dc63b27e0b19" /> | <img width="2266" height="1488" alt="Simulator Screenshot - iPad mini (A17 Pro) - US store - 2025-08-31 at 21 32 59" src="https://github.com/user-attachments/assets/b8e60179-bee7-4235-b7b9-edba83e7aa07" /> |

| Hardware main light | Hardware main dark |
|--------|--------|
| <img width="2266" height="1488" alt="Simulator Screenshot - iPad mini (A17 Pro) - US store - 2025-08-31 at 21 19 14" src="https://github.com/user-attachments/assets/e2817b7f-9004-435c-8104-ce9cb19a075c" /> | <img width="2266" height="1488" alt="Simulator Screenshot - iPad mini (A17 Pro) - US store - 2025-08-31 at 21 33 03" src="https://github.com/user-attachments/assets/7e1ca12a-8777-4d91-b3f2-2b4724042865" /> | 

| Hardware nav light | Hardware nav dark |
|--------|--------|
| <img width="2266" height="1488" alt="Simulator Screenshot - iPad mini (A17 Pro) - US store - 2025-08-31 at 21 19 18" src="https://github.com/user-attachments/assets/3dfc1633-917e-4de9-bcdb-a0513e223510" /> | <img width="2266" height="1488" alt="Simulator Screenshot - iPad mini (A17 Pro) - US store - 2025-08-31 at 21 33 06" src="https://github.com/user-attachments/assets/e5a7be52-ebaf-44b6-bcb4-a0c84a5b6058" /> | 

| Help light | Help dark |
|--------|--------|
| <img width="2266" height="1488" alt="Simulator Screenshot - iPad mini (A17 Pro) - US store - 2025-08-31 at 21 35 08" src="https://github.com/user-attachments/assets/11d1a3b6-858a-4881-9a33-863b16c6a1ac" /> | <img width="2266" height="1488" alt="Simulator Screenshot - iPad mini (A17 Pro) - US store - 2025-08-31 at 21 35 17" src="https://github.com/user-attachments/assets/3753e1fe-86ce-4b5d-ae84-2e7029913fc6" /> | 

## Testing

All changes are UI-only. In POS, navigate to `...` > `Settings` and explore them in both light and dark mode.

iamgabrielma committed:
prevents automatic uppercase from Section usage

iamgabrielma committed:
update back button on nav stack

iamgabrielma committed:
remove separators store detail view

iamgabrielma committed:
update bg color settings

iamgabrielma committed:
update headers

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
update bg color hardware view

iamgabrielma committed:
header for hardware view

iamgabrielma committed:
update bg for detail destinations

iamgabrielma committed:
update list parent bg color

iamgabrielma committed:
background consistency dark mode

iamgabrielma committed:
update foreground for selection items

iamgabrielma committed:
update help view

iamgabrielma commented on WooCommerce/Classes/POS/Presentation/Settings/PointOfSaleSettingsHardwareDetailView.swift:
```diff
@@ -41,7 +48,13 @@ struct PointOfSaleSettingsHardwareDetailView: View {
                 }
+                .listRowSeparator(.hidden)
             }
+            .listStyle(.plain)
```
In general I found very tricky to adapt Lists to the POS style, most likely when we implement the next iteration of designs we want to get rid of List entirely and build the view up from zero.

staskus replied:
For item list or coupons we use `InfiniteScrollView` + `LazyVStack` + `ForEach`. Maybe this is is where we will want to go with settings, not sure.

staskus reviewed (approved):
✅ In general, looks nice! And fairly close to the i1 of designs that was recently shared. I shared two observations on the UI from my side.

### Loading State - Store Information

Loading is shown within a card design while there's no cart design after loading. We could just show a shimmering line on a text while loading is happening. 

https://github.com/user-attachments/assets/0597d392-6027-49ef-895e-45dde117cfe3

### Header Alignment

Visually, both navigation views not aligning don't look good, it should stay in the same position as the title for consistency.

<img width="1108" height="192" alt="image" src="https://github.com/user-attachments/assets/dd57805b-c03c-4658-b94f-157a7f84dd8a" />

iamgabrielma committed:
remove shimmering card, instead add line

iamgabrielma committed:
update subnavigation headers to align with parent and sidebar

iamgabrielma commented:
Thanks for the review @staskus !

> Loading is shown within a card design while there's no cart design after loading. We could just show a shimmering line on a text while loading is happening.

True that, updated here: https://github.com/woocommerce/woocommerce-ios/pull/16065/commits/df81a6fca1159b1613114c7301e59316fe2e6ce1

> Visually, both navigation views not aligning don't look good, it should stay in the same position as the title for consistency.

Ah, good catch. I updated it here to use the POSHeader aswell, rather than the toolbar API here https://github.com/woocommerce/woocommerce-ios/pull/16065/commits/87a46da366f71e3a005b1b66ecf4cccd7467fbb7 which made the view less messier regarding modifiers:

https://github.com/user-attachments/assets/0fb37ee9-b92d-4430-9329-1a18efd235fe

> For item list or coupons we use InfiniteScrollView + LazyVStack + ForEach. Maybe this is is where we will want to go with settings, not sure.

Thanks! I'll experiment with that this week, see if we want to update it now or later 👍

staskus commented:
@iamgabrielma, thank you for the updates, looks good!
